test/system/config/test.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+set -eu -o pipefail
+
+# test compile and execution of a cupla program using its header-only variant
+# $1 C++ compiler name
+
+test_code_dir=$(dirname $0)
+
+# compiles the test case for the given accelerator
+# $1 flag if the compiled binary should be executed, 1 == execute binary else do not test binary 
+# $2 compiler name
+# $3 accelerator name (CpuOmp2Blocks, CpuOmp2Threads, CpuSerial, CpuThreads, GpuCudaRt, GpuHipRt)
+# $4 additional compiler flags
+function compile {
+    execute_bin=$1
+    compiler_name=$2
+    acc_name=$3
+    if [ $# -eq 4 ] ; then
+        compiler_flags="$4"
+    fi
+    ret=$(${compiler_name} ${test_code_dir}/main.cpp ${test_code_dir}/kernel.cpp  -I${test_code_dir}/../../../include -I${test_code_dir}/../../../alpaka/include -std=c++11 -DCUPLA_ACC_${acc_name} -DCUPLA_HEADER_ONLY -o ${acc_name} ${compiler_flags} && \
+        { echo 0; } || { echo 1; })
+    if [ $ret -eq 0 ] && [ $execute_bin -eq 1 ] ; then
+        ret=$(./$acc_name && { echo 0; } || { echo 1; })
+    fi
+    if [ $ret -ne 0 ] ; then
+        echo "Config header test failed for accelerator '$acc_name' with compiler flags '$compiler_flags'" >&2
+    fi
+    return $ret
+}
+
+boost_include="-I$BOOST_ROOT/include"
+
+compile 1 "$1" CpuOmp2Blocks "-fopenmp $boost_include"
+compile 1 "$1" CpuOmp2Blocks "-fopenmp -DCUPLA_STREAM_ASYNC_ENABLED=1 $boost_include"
+compile 0 "$1" CpuOmp2Threads "-fopenmp $boost_include"
+compile 0 "$1" CpuOmp2Threads "-fopenmp -DCUPLA_STREAM_ASYNC_ENABLED=1 $boost_include"
+compile 1 "$1" CpuSerial "$boost_include"
+compile 1 "$1" CpuSerial "-lpthread -DCUPLA_STREAM_ASYNC_ENABLED=1 $boost_include"
+compile 0 "$1" CpuThreads "-lpthread $boost_include"
+compile 0 "$1" CpuThreads "-lpthread -DCUPLA_STREAM_ASYNC_ENABLED=1 $boost_include"
+
+nvcc_found=$(which nvcc >/dev/null && { echo 0; } || { echo 1; })
+if [ $nvcc_found -eq 0 ] ; then
+    compile 0 "nvcc" GpuCudaRt "$boost_include"
+    compile 0 "nvcc" GpuCudaRt "-DCUPLA_STREAM_ASYNC_ENABLED=1 $boost_include"
+else 
+    echo "skip GpuCudaRt: nvcc not found" >&2
+fi
+
+hipcc_found=$(which hipcc >/dev/null && { echo 0; } || { echo 1; })
+if [ $hipcc_found -eq 0 ] ; then
+    rocm_root="/opt/rocm"
+    hip_include="-I${rocm_root} -I${rocm_root}/rocrand/hiprand/include/ -I${rocm_root}/rocrand/include/"
+    compile 1 "hipcc" GpuHipRt "$hip_include $boost_incude"
```

Comment:
it is `$boost_include` (it was not only the underscore, but also the missing `l`).